Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -322,22 +310,123 @@ public void Exit()
             _exitTriggered = true;
         }
 
+        private Order GetOrderById(int id)
+        {
+            Order order = null;
+
+            if (_orders.TryGetValue(id, out order) == false)
+            {
+                var clientOrder = _algorithm.Transactions.GetOrderById(id);
+
+                if (order != null)
```

Comment:
I think this should be testing `clientOrder` for null, as the `order` is gauranteed to have a value via `_orders.TryGetValue(id, out order)`